### PR TITLE
Add leave open group button in conversation

### PR DESCRIPTION
**Pull request by hjubb**

### CODE REVIEW

**Review by ceokot — APPROVED** [review]

**Comment by KeeJef** [comment]
Whats the status on this?

**Comment by hjubb** [comment]
we can probably merge this one in, just held off for release because it wasn't tied to any particular current issue so didn't want to introduce anything that might cause an issue if it wasn't necessary

File: app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt [hunk 1]
```diff
@@ -983,6 +984,24 @@ class ConversationActivityV2 : PassphraseRequiredActionBarActivity(), InputBarDe
         }
     }
 
+    override fun leaveOpenGroup(thread: Recipient) {
+        if (!thread.isOpenGroupRecipient) return
+
+        val builder = AlertDialog.Builder(this)
+            .setTitle(R.string.ConversationActivity_leave_group)
+            .setCancelable(true)
```
Comment (bemusementpark) [inline app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt:992]: `cancelable(true)` is the default. delete.

File: app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt [hunk 2]
```diff
@@ -983,6 +984,24 @@ class ConversationActivityV2 : PassphraseRequiredActionBarActivity(), InputBarDe
...
+            .setMessage(R.string.ConversationActivity_are_you_sure_you_want_to_leave_this_group)
+            .setPositiveButton(R.string.yes) { _, _ ->
+                val threadId = DatabaseComponent.get(this).threadDatabase().getThreadIdIfExistsFor(thread)
+                if (threadId == -1L) return@setPositiveButton
+                val v2OpenGroup = DatabaseComponent.get(this).lokiThreadDatabase().getOpenGroupChat(threadId) ?: return@setPositiveButton
+                OpenGroupManager.delete(v2OpenGroup.server, v2OpenGroup.room, this)
+                finish()
```
Comment (bemusementpark) [inline app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt:999]: could do something like this to remove the `return@`s if you want.

```
DatabaseComponent.get(this).threadDatabase().getThreadIdIfExistsFor(thread)
    ?.takeIf { it != -1L }
    ?.let { DatabaseComponent.get(this).lokiThreadDatabase().getOpenGroupChat(it) }
    ?.apply {
        OpenGroupManager.delete(server, room, this)
        finish()
    }
}
```

**Comment by KeeJef** [comment]
My understanding is that this task needs to be completed as part of post-release chunk 1. We might want to cherrypick this when the time comes, or simply close it and redo it when we start working on that chunk.

**Comment by mpretty-cyro** [comment]
We are likely going to look at doing the `Post Release Chunk 1` work in Compose so are going to have to reimplement this

**Pull request closed without merging**